src/NUnitEngine/nunit.engine/Agent/AgentProcess.CreateStartInfo.cs

```diff
@@ -0,0 +1,139 @@
+﻿// ***********************************************************************
```

Comment:
Nice extraction! I'm presuming all this class is just copied directly over and doesn't need reviewing!

Reply:
The class used to create the process rather than just the start info. I pulled that logic out which necessitated rearranging most of this, actually. I'll see if I can insert a commit showing the diff.

Reply:
Figuring out how to retroactively preserve the diff in source control is a bit tricky, but I can do it.
Is it good enough for me to link https://github.com/nunit/nunit-console/pull/576/commits/520d6c5c280ae1cdb6d45f97416edb928177f999#diff-d00b5d3587d1cbdc42a626eb65d5513aL157? Enough changed that an inline diff won't be as good as side-by-side anyway.

I added my trusty `StringBuilder.EscapeProcessArgument` [extension method](https://github.com/nunit/nunit-console/pull/576/files/bf9b0812ba817b540b68cbfdc3599719b02f5eda#diff-01247bb8fc9f2620caf98937cfe7ffa5R37) and [tests](https://github.com/nunit/nunit-console/pull/576/files/bf9b0812ba817b540b68cbfdc3599719b02f5eda#diff-30360af2195b50bbf97ba3cca55947e9R30).

Reply:
Yeah, looks good! I just meant I wasn't going to worry about reviewing all the code, if it was just being moved around. 🙂